File: output/ChatGPT/codenet/C++/Java/s522742913.java
import java.util.Scanner;

public class s522742913 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String source = scanner.next();

        if (source.length() > 10 || source.length() < 1) {
            return;
        }

        int max_i = (int) Math.pow(2, source.length() - 1);

        double sum = 0;

        for (int i = 0; i < max_i; ++i) {
            String plus = Integer.toBinaryString(i);

            double number = 0;

            String working = source;
            while (plus.contains("1")) {
                int index = plus.indexOf('1');
                sum += Double.parseDouble(working.substring(0, index + 1));
                plus = plus.substring(index + 1);
                working = working.substring(index + 1);
            }
            sum += Double.parseDouble(working);
        }

        System.out.printf("%.0f%n", sum);
    }
}

File: output/ChatGPT/codenet/C++/Python/s522742913.py
import math

source = input()

if len(source) > 10 or len(source) < 1:
  exit()

max_i = int(math.pow(2, len(source) - 1))

sum = 0

for i in range(max_i):
  plus = bin(i)[2:]

  number = 0

  working = source
  while any(plus):
    i = 0
    while not plus[i]:
      i += 1
    sum += float(working[:i + 1])
    plus = plus[i + 1:]
    working = working[i + 1:]
  sum += float(working)

print("{:.0f}".format(sum))